File: docs/snippets/java/benchmarking/SimpleBenchmark.java
```java title="SimpleBenchmark.java"
import io.xberg.ExtractInput;
import io.xberg.ExtractionConfig;
import io.xberg.ExtractionResult;
import io.xberg.Xberg;
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.Callable;
import java.util.concurrent.ForkJoinPool;

public final class SimpleBenchmark {
  private SimpleBenchmark() {}

  public static void main(String[] args) throws Exception {
    ExtractionConfig config = ExtractionConfig.builder()
      .withUseCache(false)
      .build();

    String filePath = "document.pdf";
    ExtractInput input = ExtractInput.fromUri(filePath);
    int numRuns = 10;

    System.out.println("Sequential extraction (" + numRuns + " runs):");
    long start = System.nanoTime();
    for (int i = 0; i < numRuns; i++) {
      Xberg.extract(input, config);
    }
    double sequentialDuration = (System.nanoTime() - start) / 1_000_000_000.0;
    double avgSequential = sequentialDuration / numRuns;
    System.out.println("  - Total time: " + String.format("%.3f", sequentialDuration) + "s");
    System.out.println("  - Average: " + String.format("%.3f", avgSequential) + "s per extraction");

    System.out.println("\nParallel extraction (" + numRuns + " runs):");
    List<Callable<ExtractionResult>> tasks = new ArrayList<>();
    for (int i = 0; i < numRuns; i++) {
      tasks.add(() -> Xberg.extract(input, config));
    }

    start = System.nanoTime();
    ForkJoinPool.commonPool().invokeAll(tasks);
    double parallelDuration = (System.nanoTime() - start) / 1_000_000_000.0;
    System.out.println("  - Total time: " + String.format("%.3f", parallelDuration) + "s");
    System.out.println("  - Average: " + String.format("%.3f", parallelDuration / numRuns) + "s per extraction");
    System.out.println("  - Speedup: " + String.format("%.1f", sequentialDuration / parallelDuration) + "x");

    ExtractionConfig cacheConfig = ExtractionConfig.builder()
      .withUseCache(true)
      .build();

    System.out.println("\nFirst extraction (populates cache)...");
    start = System.nanoTime();
    Xberg.extract(input, cacheConfig);
    double firstDuration = (System.nanoTime() - start) / 1_000_000_000.0;
    System.out.println("  - Time: " + String.format("%.3f", firstDuration) + "s");

    System.out.println("Second extraction (from cache)...");
    start = System.nanoTime();
    Xberg.extract(input, cacheConfig);
    double cachedDuration = (System.nanoTime() - start) / 1_000_000_000.0;
    System.out.println("  - Time: " + String.format("%.3f", cachedDuration) + "s");
    System.out.println("  - Cache speedup: " + String.format("%.1f", firstDuration / cachedDuration) + "x");
  }
}
```

File: docs/snippets/python/benchmarking/simple_benchmark.py
```python title="simple_benchmark.py"
import asyncio
import time

from xberg import ExtractInput, ExtractionConfig, extract


async def benchmark_extractions():
    config = ExtractionConfig(use_cache=False)
    input = ExtractInput.from_uri("document.pdf")
    num_runs = 10

    start = time.perf_counter()
    for _ in range(num_runs):
        await extract(input, config)
    sequential_duration = time.perf_counter() - start
    avg_sequential = sequential_duration / num_runs

    print(f"Sequential extraction ({num_runs} runs):")
    print(f"  - Total time: {sequential_duration:.3f}s")
    print(f"  - Average: {avg_sequential:.3f}s per extraction")

    start = time.perf_counter()
    tasks = [extract(input, config) for _ in range(num_runs)]
    await asyncio.gather(*tasks)
    parallel_duration = time.perf_counter() - start

    print(f"\nParallel extraction ({num_runs} runs):")
    print(f"  - Total time: {parallel_duration:.3f}s")
    print(f"  - Average: {parallel_duration / num_runs:.3f}s per extraction")
    print(f"  - Speedup: {sequential_duration / parallel_duration:.1f}x")

    cache_config = ExtractionConfig(use_cache=True)

    print("\nFirst extraction (populates cache)...")
    start = time.perf_counter()
    await extract(input, cache_config)
    first_duration = time.perf_counter() - start
    print(f"  - Time: {first_duration:.3f}s")

    print("Second extraction (from cache)...")
    start = time.perf_counter()
    await extract(input, cache_config)
    cached_duration = time.perf_counter() - start
    print(f"  - Time: {cached_duration:.3f}s")
    print(f"  - Cache speedup: {first_duration / cached_duration:.1f}x")


if __name__ == "__main__":
    asyncio.run(benchmark_extractions())
```
